Community/GenericPresence/automation/jsr223/python/community/genericpresence/presence.py

```diff
@@ -0,0 +1,159 @@
+    Group:Switch:OR(ON,OFF) gPresent
+    Switch vPresent
+    Switch aPresenceOverride (gPresent)
+    // Rich
+    Group:Switch:OR(ON,OFF) gRichPresent (gPresent)
+    Switch vRichPhone (gRichPresent)
+    Switch vRichPhone_Manticore_Net (gRichPresent)
+    // Jenn
+    Group:Switch:OR(ON,OFF) gJennPresent (gPresent)
+    Switch vJennPhone (gJennPresent)
+    Switch vJennPhone_Manticore_Net (gJennPresent)
+
+Notice how the there are two sensors for each person which aggregate to a
+Group. The two person Groups are made members of gPresent. There is an override
+Switch to manually set present to ON.
+"""
+from core.rules import rule
+from core.triggers import when
+from configuration import presence_config
+from core.actions import ScriptExecution
+from org.joda.time import DateTime
+
+presence_timers = {}
+
+def trigger_generator(presence_config):
+    """
+    Called to generate triggers for all of the defined presence sensors imported
+    from configuration.
+
+    Arguments:
+        - presence_config: A dict whose keys are the names of the sensor Items
+        that are managed by this Rule.
+    """
+    def generated_triggers(function):
+        for sensor in list(presence_config):
+            when("Item {} changed".format(sensor))(function)
+        return function
+    return generated_triggers
+
+def away(log, events, proxy_name, flap_time):
+    """
+    Called when a person has been detected away for more than the flapping time.
+    Log out the away state and command the Item to OFF.
+
+    Arguments:
+        - log: Logger passed in from the presence Rule.
+        - events: Access to the events Object so we can call sendCommand.
+        - proxy_name: Name of the proxy Item that we want to set to away.
+        - flap_time: Number of minutes we wait before marking the person away.
+    """
+    log.info("{} has been away for {} minutes, setting to away."
+            .format(proxy_name,flap_time))
+    events.sendCommand(proxy_name, "OFF")
+
+@rule("Presence",
+      description=("Aggregates presence sensor readings and implements "
+                   "anti-flapping for away."),
+      tags=["presence"])
+@trigger_generator(presence_config)
+def presence(event):
+    """
+    Rule that implements presence detection based on the state of sensors. The
+    Rule assumes there is a proxy Item associated with each sensor which
+    represents the presence for that sensor. There is flapping logic which
+    requires the user to be away for some minutes before the proxy will be set
+    to OFF.
+    """
+
+    # Get the proxy Item associated with the sensor.
+    try:
+        sensor_name = event.itemName
+        proxy_name = presence_config[sensor_name][0]
+        flap_time = presence_config[sensor_name][1]
+
+        # Item and sensor are the same, cancel the flapping timer if there is
+        # one.
+        global presence_timers # For the love of Pete why?! I get "use before assigned" errors without the global
```

Comment:
Because you are assigning the value at line 136. It looks like a mistake, you should be doing `presence_timers[proxy_name]` I think

Reply:
Ah, yes that would explain it. You are right, that is totally a bug. Wonder why this worked when I tested it? I bet I only tested it with one Timer running at a time. I'll test it some more this evening and check in the fix.

Reply:
OK, this bug should be fixed in the most recent commit. I was able to test live data with multiple Timers yesterday and it work as expected.